runtime/src/bank.rs

```diff
@@ -5103,23 +5103,24 @@ impl Bank {
             }
         }
 
-        let mut program_cache = self.transaction_processor.program_cache.write().unwrap();
-        program_cache.latest_root_slot = self.slot();
-        program_cache.latest_root_epoch = self.epoch();
-        program_cache.environments.program_runtime_v1 = Arc::new(
-            create_program_runtime_environment_v1(
-                &self.feature_set,
-                &self.compute_budget().unwrap_or_default(),
-                false, /* deployment */
-                false, /* debugging_features */
-            )
-            .unwrap(),
-        );
-        program_cache.environments.program_runtime_v2 =
-            Arc::new(create_program_runtime_environment_v2(
-                &self.compute_budget().unwrap_or_default(),
-                false, /* debugging_features */
-            ));
+        self.transaction_processor
+            .configure_program_runtime_environments(
+                self.slot(),
```

Comment:
slot and epoch is already available in the uninitialized `TransactionBatchProcessor`. Maybe it makes sense to use that internally instead of passing a new value here, otherwise it enables a usecase which we don't really care about.

Reply:
Wait, sorry I'm rebasing it since Lucas had some feedback on another PR about cache write locks.

Reply:
My bad, done!

Reply:
Actually I liked the other method, since it was encapsulating `program_cache` update internally. Setting `slot` and `epoch` here in program_cache here has the same issue, as it opens up a usecase where these values could be different than the ones in `TransactionBatchProcessor`. 

My suggestion was to remove `slot` and `epoch` from the now deleted `configure_` method, and use the `slot` and `epoch` that's already available in `TransactionBatchProcessor`.

Reply:
Makes sense, although technically this is no different than the original implementation in bank here at this site.

The issue with the `configure_` method I deleted is that we have a few call sites (like this one) where we only need to configure the loaders, not the fork graph. So we would probably need to have separate methods on the batch processor for configuring fork graph and loaders.

@LucasSte pointed out that having two methods forces two cache write-locks for those call sites that need to set both fork graph and loaders, which is excessive.

I guess I could just do one single `configure_` method that does what `new` does, and just pass the fork graph again here, but I don't think it's available to a `Bank` instance, only `BankForks`.